### PR TITLE
Nullable ObjectMap.m file

### PR DESCRIPTION
**Pull request by wmtylerdavis**

### CODE REVIEW

File: WPObjectMapper/NSObject+ObjectMap.m [hunk 1]
```diff
@@ -39,9 +39,10 @@
 @protocol NSObjectMappable
 @optional
 + (NSDictionary *) mapping;
-+ (instancetype) objectWithDictionary: (NSDictionary *) dictionary;
++ (nullable instancetype) objectWithDictionary: (NSDictionary *) dictionary;
 @end
 
+NS_ASSUME_NONNULL_BEGIN
```
Comment (chadmoone) [inline WPObjectMapper/NSObject+ObjectMap.m:45]: I don't think this is right.  I think the following all must be nullable:

```objc
- (instancetype)initWithJSONData:(NSData *)data{

- (instancetype)initWithXMLData:(NSData *)data{

- (instancetype)initWithSOAPData:(NSData *)data{

- (instancetype)initWithObjectData:(NSData *)data type:(CAPSDataType)type {

+ (NSArray *)arrayOfType:(Class)objectClass FromJSONData:(NSData *)data {

+(id)objectOfClass:(Class)objectClass fromXML:(NSString *)xml {

-(id)getNodeValue:(NSString *)node fromXML:(NSString *)xml {

+ (id)objectOfClass:(Class)objectClass fromJSONData:(NSData *)jsonData {

+ (NSDate *) parseDate:(id) dateValue {

-(NSString *)classOfPropertyNamed:(NSString *)propName {

-(NSString *)nameOfClass {

-(NSString *)typeFromProperty:(objc_property_t)property {

-(NSData *)JSONData{

-(NSString *)JSONString{

-(NSData *)SOAPData{

-(NSData *)XMLData{

+ (NSString *)encodeBase64WithData:(NSData *)objData {

```

I may be missing some, so please double-check me on this.

**Review by chadmoone — CHANGES_REQUESTED** [review]
Unfortunately, I think there is a bunch more which is nullable but now being declared as non-null.

**Review by chadmoone — APPROVED** [review]

File: WPObjectMapper/NSObject+ObjectMap.h [hunk 2]
```diff
@@ -61,7 +61,7 @@ typedef NS_ENUM(NSInteger, CAPSDataType) {
  
  @return The newly-initialized object.
  */
-- (instancetype)initWithJSONData:(NSData *)data;
+- (nullable instancetype)initWithJSONData:(NSData *)data;
```
Comment (vgritsenko) [inline WPObjectMapper/NSObject+ObjectMap.h:64]: No nullability specifiers on arguments, here and below. Why is so?

File: WPObjectMapper/NSObject+ObjectMap.m [hunk 3]
```diff
@@ -39,35 +39,36 @@
...
```
Comment (vgritsenko) [inline WPObjectMapper/NSObject+ObjectMap.m:45]: Why `NS_ASSUME_NONNULL_BEGIN` in the implementation file, and there is none in the header file, where it should be?

Comment (wmtylerdavis) [reply WPObjectMapper/NSObject+ObjectMap.m:45]: You're free to open a PR and fix. Thanks.

Comment (vgritsenko) [reply WPObjectMapper/NSObject+ObjectMap.m:45]: As long as I've not missed something...

Comment (wmtylerdavis) [reply WPObjectMapper/NSObject+ObjectMap.m:45]: I think you're right. I just tried to get this done as quickly as possible as we had a client downstream getting crashes.

The problem stems from Ellipsis changes on the backend, and PostKit tried to create some dumb generic RainbowItem with a dictionary.

**Pull request merged**